guidance/library/_json.py

```diff
@@ -494,9 +494,20 @@ def _gen_json_object(
     ]
     grammars = tuple(f'"{name}":' + _gen_json(json_schema=schema, definitions=definitions) for name, schema in items)
     required_items = tuple(name in required for name, _ in items)
-
+    names = set(properties.keys()) | set(required)
+    if len(names) > 0:
+        # If there are any properties, we need to disallow them as additionalProperties
+        key_grammar = as_regular_grammar(
+            And([
+                lexeme(r'"([^"\\]|\\["\\/bfnrt]|\\u[0-9a-fA-F]{4})*"'),
+                Not(lexeme('"(' + '|'.join(quote_regex(name) for name in names) + ')"')),
+            ]),
+            lexeme = True,
+        )
+    else:
+        key_grammar = _gen_json_string()
```

Comment:
Beyond the addition of new types and a bit of reorganization of `_grammar.py`, here's the meat of the bug fix.

Reply:
You could put the quotes outside the whole thing and use Or over String. Maybe a bit cleaner